README.md

```diff
@@ -27,4 +27,53 @@ The [code](https://github.com/Microsoft/vsts-urbancode-deploy) is open sourced u
 1. npm update
 1. gulp
 
-The vsix package will be produced in `_package`, and it can be uploaded to Visual Studio Team Services Marketplace for sharing.
\ No newline at end of file
+The vsix package will be produced in `_package`, and it can be uploaded to Visual Studio Team Services Marketplace for sharing.
+
+## FAQ
+### What if I need to work with Sonatype Nexus 3.x?
+Now, the best way to upload artifacts into Nexus 3.x is to use the Maven publish capabilities.
+The quickest way to deploy a file with Maven is described [here](https://maven.apache.org/plugins/maven-deploy-plugin/usage.html).
+You basically tell Maven which file to deploy and define the repository you are deploying to in the pom.xml & settings.xml files. 
+In your pom.xml file add a section describing your release and snapshot Nexus repositories. E.g.
+```
+  <repositories>
+    <repository>
+      <id>teamNexus</id>
+      <url>http://public_artifacts_server:8081/nexus/content/repositories/releases</url>
+    </repository>
+    <repository>
+      <id>teamNexus.snapshots</id>
+      <url>http://public_artifacts_server:8081/nexus/content/repositories/snapshots</url>
+      <releases>
+        <enabled>false</enabled>
+      </releases>
+      <snapshots>
+        <enabled>true</enabled>
+      </snapshots>
+    </repository>
+  </repositories>
+```
+In the settings.xml file you specify credentials, which Maven task will use to access Nexus repositories. E.g.
+```
+<?xml version="1.0" encoding="UTF-8"?>
+<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
+          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+          xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0 http://maven.apache.org/xsd/settings-1.0.0.xsd">
+  <servers>
+    <server>
+      <id>teamNexus</id>
+      <username>${nexusUsername}</username>
+      <password>${nexusPassword}</password>
+    </server>
+  </servers>
+</settings>
+```
+Note that the example uses Maven properties for user name and password credentials. 
+These properties can be set in the build task in the secure way using secure build variables.
```

Comment:
The task can be used in build and release, so make this generic - "These properties can be set in the Maven task in a secure way using secret variables in your Build/Release"